### PR TITLE
feat (api): add workflow ACL

### PR DESCRIPTION
**Pull request by sguiheux**

### CODE REVIEW

File: ui/src/app/views/workflow/show/workflow.component.ts [hunk 1]
```diff
@@ -148,6 +155,34 @@ export class WorkflowShowComponent {
...
+                    this.permFormLoading = true;
+                    this._workflowStore.addPermission(this.project.key, this.detailedWorkflow, event.gp).finally(() => {
+                        this.permFormLoading = false;
+                    }).subscribe(() => {
+                        this._toast.success('', this._translate.instant('permission_added'));
+
+                    });
+                    break;
+                case 'update':
+                    this._workflowStore.updatePermission(this.project.key, this.detailedWorkflow, event.gp).subscribe(() => {
```
Comment (bnjjj) [inline ui/src/app/views/workflow/show/workflow.component.ts]: loader ?

File: ui/src/app/views/workflow/show/workflow.component.ts [hunk 2]
```diff
@@ -148,6 +155,34 @@ export class WorkflowShowComponent {
...
+                        this._toast.success('', this._translate.instant('permission_updated'));
+                    });
+                    break;
+                case 'delete':
+                    this._workflowStore.deletePermission(this.project.key, this.detailedWorkflow, event.gp).subscribe(() => {
```
Comment (bnjjj) [inline ui/src/app/views/workflow/show/workflow.component.ts]: loader ?

File: engine/api/workflow_group.go [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+func (api *API) deleteWorkflowGroupHandler() Handler {
+	return func(ctx context.Context, w http.ResponseWriter, r *http.Request) error {
+		vars := mux.Vars(r)
+		key := vars["key"]
+		name := vars["permWorkflowName"]
+		groupName := vars["groupName"]
+
+		wf, err := workflow.Load(api.mustDB(), api.Cache, key, name, getUser(ctx))
+		if err != nil {
+			return err
```
Comment (fsamin) [inline engine/api/workflow_group.go]: wrap

File: engine/api/workflow_group.go [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+			return sdk.WrapError(err, "putWorkflowGroupHandler")
+		}
+
+		if gp.Group.Name != groupName {
+			return sdk.WrapError(sdk.ErrInvalidName, "putWorkflowGroupHandler")
+		}
+
+		wf, err := workflow.Load(api.mustDB(), api.Cache, key, name, getUser(ctx))
+		if err != nil {
+			return err
```
Comment (fsamin) [inline engine/api/workflow_group.go]: wrap

File: engine/api/workflow_group.go [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+		name := vars["permWorkflowName"]
+
+		var gp sdk.GroupPermission
+		if err := UnmarshalBody(r, &gp); err != nil {
+			return sdk.WrapError(err, "postWorkflowGroupHandler")
+		}
+
+		wf, err := workflow.Load(api.mustDB(), api.Cache, key, name, getUser(ctx))
+		if err != nil {
+			return err
```
Comment (fsamin) [inline engine/api/workflow_group.go]: wrap

File: ui/src/app/views/workflow/show/workflow.component.ts (hunk 1 above)
Comment (sguiheux) [reply ui/src/app/views/workflow/show/workflow.component.ts]: already in permission component

File: ui/src/app/views/workflow/show/workflow.component.ts (hunk 2 above)
Comment (sguiheux) [reply ui/src/app/views/workflow/show/workflow.component.ts]: already in permission component

**Pull request merged**